### PR TITLE
Add on-scroll attribute to annoref shortcode

### PR DESCRIPTION
**Pull request by hbalenda**
## Changes
- Rename `annoref` to `ref` and remove old `ref` (figureRef)
- Add scroll interactions to `ref` shortcode
- Add transitions to rotating images

## Usage
`{% ref fig="fig-1" onscroll="true" %}`

## Testing
- Checkout this branch `feature/DEV-14268_scroll-interactions`
- Checkout https://github.com/thegetty/quire-starter-iiif/pull/5

`quire new scroll-interactions ~/Path/to/quire-starter-iiif --quire-path ~/Path/to/quire/packages/11ty`

### CODE REVIEW

**Review by mphstudios — APPROVED** [review]

**Comment by hbalenda** [comment]
@geealbers I've got the vertical layout with scroll interactions set up now if you'd like to review, it works in parallel with https://github.com/thegetty/quire-starter-iiif/pull/5 (and I will update the other starters after this is finalized). Thank you!

File: packages/11ty/CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,13 @@ Changelog entries are classified using the following labels:
 
 ## [unreleased]
 
+### Added
+- Adds scroll interactions to `ref` shortcode
+- Adds rotation animations to image sequences referenced using the `ref` shortcode
+
+### Changed
+- Rename `annoref` shortcode to `ref`, remove `figureRef`.
+
```
Comment (mphstudios) [inline packages/11ty/CHANGELOG.md:29]:
```suggestion
### Added

- Scroll interactions to `ref` shortcode
- Animation to image sequences referenced using the `ref` shortcode

### Changed

- Renames `annoref` shortcode to `ref`

### Removed

- `figureRef` shortcake (replaced by `ref` shortcake)

```

File: packages/11ty/_includes/components/figure/image/image-sequence.webc [hunk 2]
```diff
@@ -309,14 +329,15 @@ img.${placeholder} {
         this.hideOverlays();
         if (this.oldX) {
           const deltaX = clientX - this.oldX;
+          const deltaIndex = Math.floor(Math.log(Math.abs(deltaX))) || 1;
```
Comment (mphstudios) [inline packages/11ty/_includes/components/figure/image/image-sequence.webc:332]:
```suggestion
          const index = Math.floor(Math.log(Math.abs(deltaX))) || 1;
```

File: packages/11ty/_includes/components/figure/image/image-sequence.webc [hunk 3]
```diff
@@ -309,14 +329,15 @@ img.${placeholder} {
...
           if (deltaX > 0) {
             this.isReversed
-              ? this.previousCanvas(deltaX)
-              : this.nextCanvas(deltaX);
+              ? this.previousCanvas(deltaIndex)
+              : this.nextCanvas(deltaIndex);
```
Comment (mphstudios) [inline packages/11ty/_includes/components/figure/image/image-sequence.webc:336]:
```suggestion
              ? this.previousCanvas(index)
              : this.nextCanvas(index);
```

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js [hunk 4]
```diff
@@ -202,21 +228,41 @@ const selectChoice = (canvasPanel, annotation) => {
...
+          historyBehavior: 'replace',
+          region,
+          sequence
+        })
+      intersectionObserverFactory(ref, callback)
+    } else {
+      ref.addEventListener('click', ({ target }) =>
+        goToFigureState({ annotationIds, figureId, region, sequence })
+      )
+    }
```
Comment (mphstudios) [inline packages/11ty/content/_assets/javascript/application/canvas-panel.js:265]: Is the following possibly called twice?
```
ref.addEventListener('click', ({ target }) =>
  goToFigureState({ annotationIds, figureId, region, sequence })
)
```

**Review by mphstudios — APPROVED** [review]

File: packages/11ty/_includes/components/figure/image/image-sequence.webc [hunk 5]
```diff
@@ -309,14 +329,15 @@ img.${placeholder} {
...
           } else if (deltaX < 0) {
             this.isReversed
-              ? this.nextCanvas(deltaX)
-              : this.previousCanvas(deltaX);
+              ? this.nextCanvas(deltaIndex)
+              : this.previousCanvas(deltaIndex);
```
[the remark this answers is not in the document]
Comment (hbalenda) [reply packages/11ty/_includes/components/figure/image/image-sequence.webc:340]: @mphstudios it is actually the `delta` - `nextCanvas` and `previousCanvas` accept the number of steps to take forward or backwards from the current index

**Review by mphstudios — APPROVED** [review]

**Review by mphstudios — APPROVED** [review]

**Pull request merged**